Difference between Binaryen.js release and debug builds regarding BigInt

Debug builds of Binaryen.js (both JS and Wasm) seem to differ from Release builds, in that Debug builds appear to anticipate BigInt integration in the C ABI while Release builds do not.

Compiling the `binaryen_js` target using `-DCMAKE_BUILD_TYPE=Debug` fails at the end of compilation during `wasm2js` invocation:
```
[100%] Linking CXX executable bin/binaryen_js.js
em++: error: wasm2js does not support WASM_BIGINT
make[3]: *** [CMakeFiles/binaryen_js.dir/build.make:393: bin/binaryen_js.js] Error 1
make[2]: *** [CMakeFiles/Makefile2:363: CMakeFiles/binaryen_js.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:370: CMakeFiles/binaryen_js.dir/rule] Error 2
make: *** [Makefile:190: binaryen_js] Error 2
emmake: error: 'make -j12 binaryen_js' failed (returned 2)
```

Compiling the `binaryen_wasm` target using `-DCMAKE_BUILD_TYPE=Debug` works, but yields a binary where `_BinaryenLiteralInt64` expects a BigInt argument, not two number arguments (low and high bits), then failing at runtime when `_BinaryenLiteralInt64` is invoked with two number values:
```
TypeError: Cannot convert [some number] to a BigInt
  at file://path/to/binaryen_wasm.js:745:22
```

Compiling either using `-DCMAKE_BUILD_TYPE=Release` produces artifacts that do not assume BigInt integration.

Origin seems to be [in CmakeLists.txt](https://github.com/WebAssembly/binaryen/blob/0578862dc5569a4b544800d666458d698536a1a5/CMakeLists.txt#L310), where `-sWASM_BIGINT` is specified for Debug builds only.

When just removing `-sWASM_BIGINT`, both the `binaryen_js` and `binaryen_wasm` targets fail to build, but with a different error:

```
[ 98%] Linking CXX executable bin/binaryen_js.js
em++: error: changes to the wasm are required after link, but disallowed by ERROR_ON_WASM_CHANGES_AFTER_LINK: ['path/to/wasm-emscripten-finalize', '-g', '--dyncalls-i64', '--pass-arg=legalize-js-interface-exported-helpers', '--dwarf', 'bin/binaryen_js.wasm', '-o', 'bin/binaryen_js.wasm']
note: to disable int64 legalization (which requires changes after link) use -sWASM_BIGINT
make[3]: *** [CMakeFiles/binaryen_js.dir/build.make:393: bin/binaryen_js.js] Error 1
make[2]: *** [CMakeFiles/Makefile2:363: CMakeFiles/binaryen_js.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:370: CMakeFiles/binaryen_js.dir/rule] Error 2
make: *** [Makefile:190: binaryen_js] Error 2
emmake: error: 'make -j12 binaryen_js' failed (returned 2)
```

When removing both `-sWASM_BIGINT` and `-sERROR_ON_WASM_CHANGES_AFTER_LINK`, both targets work, but `binaryen_js` with warnings:
```
warning: running pass 'flatten' which is not fully compatible with DWARF
warning: running pass 'flatten' which is not fully compatible with DWARF
```

Respective PR originally introducing the two flags: https://github.com/WebAssembly/binaryen/pull/4842

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -305,8 +305,19 @@ else()
 endif()
 
 if(EMSCRIPTEN)
-  # Note: to debug with DWARF you will need something like
-  #   -sWASM_BIGINT -sERROR_ON_WASM_CHANGES_AFTER_LINK
+  # Note: to debug with DWARF you will usually want to enable BIGINT support, as
+  # that helps avoid running Binaryen on the wasm after link. Binaryen's DWARF
+  # rewriting has known limitations, so avoiding it during link is recommended
+  # where possible (like local debugging).
+  #
+  # Note that this is debug info for Binaryen itself, that is, when you are
+  # debugging Binaryen source code. This flag has no impact on what Binaryen
+  # does when run on wasm files.
+  option(ENABLE_BIGINT "Enable wasm Bigint" OFF)
+  if(ENABLE_BIGINT)
+    add_link_flag("-sERROR_ON_WASM_CHANGES_AFTER_LINK")
+    add_link_flag("-sWASM_BIGINT")
+  endif()
 
   if("${CMAKE_BUILD_TYPE}" MATCHES "Release")
     # link with -O3 for metadce and other powerful optimizations. note that we

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -313,7 +313,7 @@ if(EMSCRIPTEN)
   # Note that this is debug info for Binaryen itself, that is, when you are
   # debugging Binaryen source code. This flag has no impact on what Binaryen
   # does when run on wasm files.
-  option(ENABLE_BIGINT "Enable wasm Bigint" OFF)
+  option(ENABLE_BIGINT "Enable wasm BigInt support" OFF)
   if(ENABLE_BIGINT)
     add_link_flag("-sERROR_ON_WASM_CHANGES_AFTER_LINK")
     add_link_flag("-sWASM_BIGINT")

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -305,10 +305,10 @@ else()
 endif()
 
 if(EMSCRIPTEN)
-  if("${CMAKE_BUILD_TYPE}" MATCHES "Debug")
-    add_link_flag("-sERROR_ON_WASM_CHANGES_AFTER_LINK")
-    add_link_flag("-sWASM_BIGINT")
-  else()
+  # Note: to debug with DWARF you will need something like
+  #   -sWASM_BIGINT -sERROR_ON_WASM_CHANGES_AFTER_LINK
+
+  if("${CMAKE_BUILD_TYPE}" MATCHES "Release")
     # link with -O3 for metadce and other powerful optimizations. note that we
     # must use add_link_options so that this appears after CMake's default -O2
     add_link_options("-O3")

